### PR TITLE
Fixed keyboard now showing up in open-choice dialog.

### LINKED ISSUE #1293: Soft keyboard is not opened for open choice view.
**Describe the bug**

**Component**
Core library, SDC library, reference app, or SDC gallery app

**To Reproduce**

1. Open catalog app.
2. Click on the Open choice view.
3. Click on the box.
4. Click on other option
5. Click on add another answer
6. See, it does not open soft keyboard to type the text.

**Expected behavior**

**Screenshots**

https://user-images.githubusercontent.com/86107848/163375039-7d6d961f-ff16-46d3-920e-cb4c09c38834.mov

**Smartphone (please complete the following information):**
 - Device: [e.g. Pixel4a emulator]
 - Android version: [e.g. Settings -> About phone -> Android version]
 - Build number: [e.g. Settings -> About phone -> Build number]

**Additional context**

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue.

### PR DESCRIPTION
**Pull request by aditya-07**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1293

**Description**
Android was preventing keyboard to show up in `EditText` inside a `RecyclerView` in a `AlertDialog`. 
Changes done are as: 

1. Clear dialog windows flags to allow keyboard to show up. (see https://stackoverflow.com/a/9118027)
2. Adjust soft input mode for Dialog to resize itself when keyboard shows up so that the Ok-Cancel buttons are visible.
3. Moved the input `EditText` to `TextInputLayout` .
4. Fixed deleteButton by adding tint color so that it is visible in all the applied themes.
5. Moved from default `AlertDialog.Builder` to `MaterialAlertDialogBuilder`.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Bug fix 

**Screenshots (if applicable)**

https://user-images.githubusercontent.com/18224849/163564895-f1056780-7bcd-4c92-b1c6-46f12b275334.mov

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Review by jingtang10 — APPROVED** [review]
great fix thanks @aditya-07! 👍🙏👌

also importantly this pr uses the material dialog. please add that to the title/description.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/OptionSelectDialogFragment.kt [hunk 1]
```diff
@@ -74,12 +76,31 @@ internal class OptionSelectDialogFragment(
...
+          dialog?.window?.let {
+            // Android: EditText in Dialog doesn't pull up soft keyboard
+            // https://stackoverflow.com/a/9118027
+            it.clearFlags(
+              WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE or
+                WindowManager.LayoutParams.FLAG_ALT_FOCUSABLE_IM
+            )
+            // Adjust the dialog after the keyboard is on so that OK-CANCEL buttons are visible.
+            // it.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE)
+            if (Build.VERSION.SDK_INT > 29) {
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/OptionSelectDialogFragment.kt:96]: can you also document why it's necessary to treat different API versions differently here?

**Review by jingtang10 — APPROVED** [review]
feel free to merge once the comments are addressed. thanks

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/OptionSelectDialogFragment.kt (hunk 1 above)
Comment (aditya-07) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/OptionSelectDialogFragment.kt]: Done.

**Pull request merged**